File: 198-house-robber/house-robber.js
var rob = function(nums) {
let n = nums.length;
if (n === 1)
    return nums[0];

let dp = [];
dp.push(nums[0]);
dp.push(Math.max(nums[0],nums[1]));
for(let i = 2; i < n;i++){
    dp[i] = Math.max(dp[i-1],nums[i]+dp[i-2]);
}

return dp[dp.length-1];

};

File: 198-house-robber/house-robber.cs
public class Solution {
    public int Rob(int[] nums) {
        int n = nums.Length;
        if (n == 1)
        return nums[0];
        List<int> dp = new List<int>(n);
        dp.Add(nums[0]);
        dp.Add(Math.Max(nums[1],nums[0]));
        for (int i = 2; i < n; i++){
            dp.Add(Math.Max(dp[i-1], dp[i-2]+nums[i]));
        }
        return dp[^1];
    }
}